### PR TITLE
[MODORDSTOR-356] - Initial setup of claiming batch job

### PR DESCRIPTION
**Pull request by Abdulkhakimov**
## Purpose
Forgot to point out a couple places to update the date of the status change in the previous PR: #791 .

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java [hunk 1]
```diff
@@ -400,8 +401,9 @@ protected Future<Map<String, List<Piece>>> storeUpdatedPieceRecords(Map<String,
       .ofValues(piecesGroupedByPoLine)
       .flatMap(List::stream)
       .filter(this::isSuccessfullyProcessedPiece)
-      .map((Piece piece) -> storeUpdatedPieceRecord(piece, requestContext))
-      .collect(Collectors.toList());
+      .map(piece -> piece.withStatusUpdatedDate(new Date()))
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:404]: What if status was not updated in this case, for example just some another field in piece updated?

File: src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java [hunk 2]
```diff
@@ -121,18 +122,23 @@ public Future<Void> updatePieceRecord(Piece piece, RequestContext requestContext
       .compose(v -> inventoryManager.updateItemWithPieceFields(piece, requestContext))
       .onSuccess(vVoid ->
         pieceStorageService.getPieceById(piece.getId(), requestContext).onSuccess(pieceStorage -> {
+            Piece.ReceivingStatus receivingStatusUpdate = piece.getReceivingStatus();
             Piece.ReceivingStatus receivingStatusStorage = pieceStorage.getReceivingStatus();
+            boolean isReceivingStatusChanged = receivingStatusStorage.compareTo(receivingStatusUpdate) != 0;
+
+            if(isReceivingStatusChanged) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java:129]: let's add info log here that can help us in troubleshooting

File: src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java (hunk 1 above)
Comment (Abdulkhakimov) [reply src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java:404]: checkin and receiving endpoints is used to transite piece to another status

File: src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java (hunk 2 above)
Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java:129]: we already have loggin statements in #L138 and #L139

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java:129]: basically they are with debug level and basically separate log that include pieceId and both statuses(from and to) would be useful expessionally if we have logic with conditional statements, but it up to you

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Pull request merged**